pyriemann/transfer/pipelines.py

```diff
@@ -0,0 +1,512 @@
+    """No transformation on data for transfer learning
+
+    No transformation of the data points between the domains.
+    This is what we call the Direct Center Transfer (DCT) method.
+    """
+
+    def __init__(self):
+        pass
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X, y=None):
+        return X
+
+    def fit_transform(self, X, y):
+        return self.fit(X, y).transform(X, y)
+
+
+class TLCenter(BaseEstimator, TransformerMixin):
+    """Recenter data for transfer learning
+
+    Recenter the data points from each domain to the Identity on manifold, ie
+    make the geometric mean of the datasets become the identity. This operation
+    corresponds to a whitening step if the SPD matrices represent the spatial
+    covariance matrices of multivariate signals.
+
+    Parameters
+    ----------
+    target_domain : str
+        Which domain to consider as target
+    metric : str, default='riemann'
+        The metric for mean, can be: 'ale', 'alm', 'euclid', 'harmonic',
+        'identity', 'kullback_sym', 'logdet', 'logeuclid', 'riemann',
+        'wasserstein', or a callable function.
+
+    """
+
+    def __init__(self, target_domain, metric='riemann'):
+        """Init"""
+        self.target_domain = target_domain
+        self.metric = metric
+
+    def fit(self, X, y):
+        _, _, domains = decode_domains(X, y)
+        self.recenter_ = {}
+        for d in np.unique(domains):
+            idx = domains == d
+            self.recenter_[d] = Whitening(metric=self.metric).fit(X[idx])
+        return self
+
+    def transform(self, X, y=None):
+        # Used during inference, apply recenter from specified target domain.
+        return self.recenter_[self.target_domain].transform(X)
+
+    def inverse_transform(self, X, y=None):
+        return self.recenter_[self.target_domain].inverse_transform(X)
+
+    def fit_transform(self, X, y):
+        # used during fit, in pipeline
+        self.fit(X, y)
+        _, _, domains = decode_domains(X, y)
+        X_rct = np.zeros_like(X)
+        for d in np.unique(domains):
+            idx = domains == d
+            X_rct[idx] = self.recenter_[d].transform(X[idx])
+        return X_rct
+
+
+class TLStretch(BaseEstimator, TransformerMixin):
+    """Stretch data for transfer learning
+
+    Change the dispersion of the datapoints around their geometric mean
+    for each dataset so that they all have the same desired value.
+
+    The dispersion is defined as the squared sum of the AIRM-induced distances
+    of each point with respect to their geometric mean.
+
+    Parameters
+    ----------
+    target_domain : str
+        Which domain to consider as target
+    dispersion : float, default=1.0
+        Which target value for the dispersion of the data points
+    centered_data : bool, default=False
+        Whether the data has been re-centered to the Identity beforehand
+    """
+
+    def __init__(self, target_domain, final_dispersion=1.0,
```

Comment:
Mainly because the operation for "stretch" as we know it from the RPA paper is only defined for the Riemannian distance. With that said, we could define it for the Euclid distance and for the LogEuclid ones. However, I am not sure we can do anything very easy for all of the metrics available in `pyriemann`. Anyways, do you think it is really worth defining these variations for stretch?

Reply:
Well, I've added the option of choosing any metric for calculating the dispersion of the SPD matrices. Note, however, that the stretching is still in "Riemannian" terms since an Euclidean stretch would only make sense if we re-centered the data points to zero, which is not SPD...